examples/imdb/imdb/.tensorleap/dataset.py

```diff
@@ -0,0 +1,187 @@
+    val_paths = train_dict['pos'][half_t_size:half_t_size+half_v_size] + \
+                train_dict['neg'][half_t_size:half_t_size+half_v_size]
+    val_gt = get_gt(val_paths)
+    val_metrics = train_dict['pos_metrics'][half_t_size:half_t_size+half_v_size] + \
+                  train_dict['neg_metrics'][half_t_size:half_t_size+half_v_size]
+    return train_paths, train_gt, train_metrics, val_paths, val_gt, val_metrics
+
+
+def _connect_to_gcs() -> Bucket:
+    print("_connect_to_gcs")
+    gcs_client = storage.Client(project=PROJECT_ID, credentials=AnonymousCredentials())
+    return gcs_client.bucket(BUCKET_NAME)
+
+
+def _download(cloud_file_path: str, local_file_path: Optional[str] = None) -> str:
+    BASE_PATH = "imdb"
+    cloud_file_path = join(BASE_PATH, cloud_file_path)
+    print("download")
+    # if local_file_path is not specified saving in home dir
+    if local_file_path is None:
+        home_dir = os.getenv("HOME")
+        local_file_path = os.path.join(home_dir, "Tensorleap_data", BUCKET_NAME, cloud_file_path)
+    print(local_file_path)
+    # check if file already exists
+    if os.path.exists(local_file_path):
+        return local_file_path
+
+    bucket = _connect_to_gcs()
+    dir_path = os.path.dirname(local_file_path)
+    os.makedirs(dir_path, exist_ok=True)
+    blob = bucket.blob(cloud_file_path)
+    blob.download_to_filename(local_file_path)
+    return local_file_path
+
+
+def subset_func() -> List[SubsetResponse]:
+    print("subset_func")
+    tokenizer, train_dict = download_load_assets()
+    dataset_binder.cache_container["word_to_index"]["tokens"] = tokenizer.word_index
+    max_size = 2*len(train_dict['pos'])
+    train_size = min(NUMBER_OF_SAMPLES, int(0.9*max_size))
+    half_t_size = int(train_size/2)
+    val_size = int(0.1*train_size)
+    half_v_size = int(val_size/2)
+    train_paths, train_gt, train_metrics, val_paths, val_gt, val_metrics\
+        = create_input(train_dict, half_t_size, half_v_size)
+    train = SubsetResponse(length=2*half_t_size, data={'paths': train_paths, 'gt': train_gt, 'tokenizer': tokenizer,
+                                                       'metrics': train_metrics})
+    val = SubsetResponse(length=2*half_v_size, data={'paths': val_paths, 'gt': val_gt, 'tokenizer': tokenizer,
+                                                     'metrics': val_metrics})
+    response = [train, val]
+    return response
+
+
+def input_tokens(idx: int, subset: SubsetResponse) -> np.ndarray:
+    print("input_tokens")
+    comment_path = subset.data['paths'][idx]
+    local_path = _download(comment_path)
+    with open(local_path, 'r') as f:
+        comment = f.read()
+    tokenizer = subset.data['tokenizer']
+    padded_input = prepare_input(tokenizer, comment)
+    return padded_input
+
+
+def gt_sentiment(idx: int, subset: Union[SubsetResponse, list]) -> List[float]:
+    print("gt_sentiment")
```

Comment:
Do you think that we need the prints in the dataset function here?

Reply:
all prints removed